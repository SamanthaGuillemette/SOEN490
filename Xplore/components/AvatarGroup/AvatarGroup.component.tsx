import { View } from "../View";
import { Avatar } from "../Avatar";
import styles from "./AvatarGroup.styles";

interface AvatarGroupProps {
  usersAvatars: string[];
  scale?: number;
}

export const AvatarGroup = ({ scale = 3, ...props }: AvatarGroupProps) => {
  const avatarCount = props.usersAvatars.length - 3;
  return (
    <View>
      {props.usersAvatars.length > 4 ? (
        <View style={styles.groupAvatarContainer}>
          {props.usersAvatars.slice(0, 3).map((avatar, key) => (
            <Avatar
              key={key}
              name="Username"
              imageURL={avatar}
              size={scale * 30}
              style={styles.singleAvatar}
            />
          ))}
          <Avatar
            name="+"
            avatarCount={avatarCount}
            size={scale * 20}
            style={styles.singleAvatar}
          />
        </View>
      ) : (
        <View style={styles.groupAvatarContainer}>
          {props.usersAvatars.map((avatar, key) => (
            <Avatar
              key={key}
              name="Username"
              imageURL={avatar}
              size={scale * 20}
              style={styles.singleAvatar}
            />
          ))}
        </View>
      )}
    </View>
  );
};
