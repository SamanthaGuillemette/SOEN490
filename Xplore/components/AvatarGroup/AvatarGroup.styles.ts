import { StyleSheet } from "react-native";

export default StyleSheet.create({
  groupAvatarContainer: {
    flexDirection: "row",
    alignItems: "center",
  },
  singleAvatar: {
    marginLeft: -15,
  },
});
