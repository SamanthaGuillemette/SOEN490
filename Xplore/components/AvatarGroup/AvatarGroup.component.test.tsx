import { cleanup } from "react-native-testing-library";
import renderer from "react-test-renderer";
import React from "react";
import { AvatarGroup } from "./AvatarGroup.component";

afterAll(() => {
  cleanup();
});

describe("AvatarGroup should render correctly", () => {
  it("should render less than 4 avatars correctly", () => {
    const avatarCount = [
      { avatar: "", username: "Testing123" },
      { avatar: "", username: "Testing123" },
    ];
    const avatars = renderer.create(
      <AvatarGroup usersAvatars={avatarCount} />
    ).toJSON;
    expect(avatars).toMatchSnapshot();
  });
});

describe("AvatarGroup should change after rendering more than 4 avatars", () => {
  it("should render less than 5 avatars correctly", () => {
    const avatarCount = [
      { avatar: "", username: "Testing123" },
      { avatar: "", username: "Testing123" },
      { avatar: "", username: "Testing123" },
      { avatar: "", username: "Testing123" },
      { avatar: "", username: "Testing123" },
    ];
    const avatars = renderer.create(
      <AvatarGroup usersAvatars={avatarCount} />
    ).toJSON;
    expect(avatars).toMatchSnapshot();
  });
});
