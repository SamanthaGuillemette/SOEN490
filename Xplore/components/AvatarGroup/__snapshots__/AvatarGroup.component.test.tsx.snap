// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarGroup should render correctly should render correctly 1`] = `[Function]`;
